packages/core/src/enqueue_links/shared.ts

```diff
@@ -79,6 +161,10 @@ export function createRequestOptions(sources: (string | Record<string, unknown>)
             } catch (err) {
                 return false;
             }
+        })
+        .map((requestOptions) => {
+            if (!requestOptions.userData) requestOptions.userData = {};
```

Comment:
we can do better now :]

```suggestion
            requestOptions.userData ??= {};
```